apps/front/src/components/ui/formInput/formInput.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+    IconButton,
+    InputAdornment,
+    TextField,
+    TextFieldProps,
+} from '@mui/material'
+import React, { FC } from 'react'
+import { Controller, get, useFormContext } from 'react-hook-form'
+import { Visibility, VisibilityOff } from '@mui/icons-material'
+import useFormInput from '@src/components/ui/formInput/useFormInput'
+
+type FormInputProps = {
+    name: string
+    isPassword?: boolean
```

Comment:
Zapoznaj się z zasadami SOLID w react to pomogą w organizacji kodu
https://www.youtube.com/watch?v=t_h_A6RkM7A&ab_channel=CoderOne 
Taki `isPassword` łamie zasadę open-closed i przykładowo, gdybyś miał dodać nowy props, który w podobny sposób modyfikuje kod to w tym komponencie by się pojawiło mnóstwo warunków. np
taki kod:
`type={showPassword || !isPassword ? 'text' : 'password'}`
mógłby się przerodzić w coś takiego:
`type={showPassword || !isPassword ? isCondition ? 'number' : 'text' : 'password'}`
Co by utraciło na czytelności i byłoby podatne na błędy